make icons directorylisting configurable

nbeernink opened the pull request:
This should allow people to disable directory index on the icons/ dir via an override.

mhaskel commented:
@nbeernink thanks for the contribution, and sorry it took so long for us to review it. I have a couple of comments on your PR, but I'll add them inline. Thanks again!

mhaskel commented on manifests/mod/alias.pp:
```diff
@@ -6,6 +6,7 @@
     'redhat'  => '/var/www/icons',
     'freebsd' => '/usr/local/www/apache22/icons',
   }
+  $icons_index = 'Indexes Multiviews'
```
I think this should be added as a parameter for `apache::mod::alias` if you actually want to be able to set it.

mhaskel commented on templates/mod/alias.conf.erb:
```diff
@@ -1,7 +1,7 @@
 Alias /icons/ "<%= @icons_path %>/"
 <Directory "<%= @icons_path %>">
-    Options Indexes MultiViews
+    <%= @icons_index %>
```
Should this be `Options <%= @icons_index %>` ?

nbeernink commented:
I patched the code but ran out of time trying to solve the problem of not being able to push to this PR. Will try again later.

ekohl commented on manifests/mod/alias.pp:
```diff
@@ -11,6 +12,7 @@
     },
     'freebsd' => '/usr/local/www/apache22/icons',
   }
+  $icons_options = $icons_options
```
Isn't this redundant?

nbeernink replied:
Ah you're right.

ekohl commented:
Test failures are unrelated and should be fixed with https://github.com/puppetlabs/puppetlabs-concat/pull/263

mhaskel commented on manifests/mod/alias.pp:
```diff
@@ -1,5 +1,6 @@
 class apache::mod::alias(
-  $apache_version = $apache::apache_version
+  $apache_version = $apache::apache_version,
+  $icons_options  = 'Indexes Multiviews',
```
This should be `'Indexes MultiViews'`

mhaskel commented:
@nbeernink I commented above on one more code update to not change default behavior. Could you also update unit tests [here](https://github.com/puppetlabs/puppetlabs-apache/blob/master/spec/classes/mod/alias_spec.rb) to exercise the new parameter, and update the README to indicate that `apache::mod::alias` has parameters that can be set (should just be adding a `*`). Thanks!

nbeernink commented:
@mhaskel, added the docs and corrected the multiviews option, but I don't know how I would write the tests you require. Hope you could point me in the right direction or add them for me. :)

igalic commented:
@nbeernink n.b.: you also need to [rebase](http://git-scm.com/book/en/Git-Branching-Rebasing) this pr now…

as for tests, take a look at other [spec tests for mod_alias](https://github.com/puppetlabs/puppetlabs-apache/blob/master/spec/classes/mod/alias_spec.rb)

mhaskel commented:
@nbeernink you'll need to add a new block in the tests where you add a :params block like https://github.com/puppetlabs/puppetlabs-apache/blob/master/spec/classes/mod/auth_cas_spec.rb#L4-L9 and check that the file contains the values you're setting (like https://github.com/puppetlabs/puppetlabs-apache/blob/master/spec/classes/mod/authnz_ldap_spec.rb#L26-L28)
